libsnowflakeclient/include/snowflake_client.h

```diff
@@ -81,7 +81,8 @@ typedef enum sf_error_code {
     SF_ERROR_BAD_JSON,
     SF_ERROR_RETRY,
     SF_ERROR_CURL,
-    SF_ERROR_BAD_ATTRIBUTE_TYPE
+    SF_ERROR_BAD_ATTRIBUTE_TYPE,
+    SF_ERROR_APPLICATION_ERROR
```

Comment:
ok for now. Will revisit the set of error code later to align with Snowflake error code.